elasticgraph-graphql/spec/integration/elastic_graph/graphql/datastore_query/filtering_spec.rb

```diff
@@ -1007,6 +1007,154 @@ def search_datastore(**options, &before_msearch)
         expect(results).to match_array(ids_of(widget1))
       end
 
+      specify "`matches_query_with_prefix` performs bool prefix matching with exact term matching" do
+        index_into(
+          graphql,
+          widget1 = build(:widget, description_text: "red widget small"),
+          widget2 = build(:widget, description_text: "red widget medium")
+        )
+
+        results = search_with_filter(
+          "description_text",
+          "matches_query_with_prefix",
+          {
+            "query_with_prefix" => "re",
+            "allowed_edits_per_term" => enum_value("MatchesQueryAllowedEditsPerTermInput", "NONE"),
+            "require_all_terms" => true
+          }
+        )
+
+        expect(results).to match_array(ids_of(widget1, widget2))
+      end
+
+      specify "`matches_query_with_prefix` performs bool prefix matching with fuzzy term matching" do
+        index_into(
+          graphql,
+          widget1 = build(:widget, description_text: "red widget small")
```

Comment:
Here too we need some counter examples of widgets which don't match to prove that the implementation is actually doing something besides returning all documents.